src/sync-merkleroots.test.ts

```diff
@@ -0,0 +1,315 @@
+import fetchMock from 'jest-fetch-mock';
+import {
+  ErrorStaleLastEvaluatedKey,
+  ErrorSyncMerkleRootsTimeout,
+  ExclusiveStartKeyPage,
+} from '../dist/typescript-npm-package.cjs';
+import { SpvWalletClient } from './client';
+import { MerkleRoot, MerkleRootsRepository } from './types';
+
+const server = 'http://localhost:3003/api/v1';
+
+const client = new SpvWalletClient(server, {
+  xPriv:
+    'xprv9s21ZrQH143K3axKPtYBDKsrAvN3J85z6nZuW5ihYK8JpDWfqHdjswvUnbXzUpMBne1WD6FQmTzymB4Pt3u3UVSauxzq5PswBWr3vYtowmW',
+});
+
+// creates a simulated repository a client passes to SyncMerkleRoots()
+const createRepository = (dataBase: { merkleRoots: MerkleRoot[] }): MerkleRootsRepository => {
```

Comment:
Why not use simply `merkleRoots: MerkleRoot[]`? Because of this is an array - you're passing only reference - not copy.
```suggestion
const createRepository = (merkleRootsDB: MerkleRoot[]): MerkleRootsRepository => {
```